shell/src/main/java/alluxio/cli/fs/command/CheckConsistencyCommand.java

```diff
@@ -122,43 +151,85 @@ private void runConsistencyCheck(AlluxioURI path, boolean repairConsistency) thr
       }
     } else {
       Collections.sort(inconsistentUris);
-      System.out.println(path + " has: " + inconsistentUris.size() + " inconsistent files.");
-      List<AlluxioURI> inconsistentDirs = new ArrayList<AlluxioURI>();
+      System.out.println(String.format("%s has: %d inconsistent files. Repairing with %d threads.",
+          path, inconsistentUris.size(), repairThreads));
+      ConcurrentHashSet<AlluxioURI> inconsistentDirs = new ConcurrentHashSet<>();
+
+      ExecutorService svc = Executors.newFixedThreadPool(repairThreads);
+      CompletionService<Boolean> completionService = new ExecutorCompletionService<>(svc);
+      ConcurrentHashSet<Exception> exceptions = new ConcurrentHashSet<>();
+
+      int totalUris = inconsistentUris.size();
       for (AlluxioURI inconsistentUri : inconsistentUris) {
-        URIStatus status = mFileSystem.getStatus(inconsistentUri);
-        if (status.isFolder()) {
-          inconsistentDirs.add(inconsistentUri);
-          continue;
-        }
-        System.out.println("repairing path: " + inconsistentUri);
-        DeletePOptions deleteOptions = DeletePOptions.newBuilder().setAlluxioOnly(true).build();
-        mFileSystem.delete(inconsistentUri, deleteOptions);
-        mFileSystem.exists(inconsistentUri);
-        System.out.println(inconsistentUri + " repaired");
-        System.out.println();
+        completionService.submit(() -> {
+          try {
+            URIStatus status = mFileSystem.getStatus(inconsistentUri);
+            if (status.isFolder()) {
+              inconsistentDirs.add(inconsistentUri);
+              return;
+            }
+            System.out.println("repairing path: " + inconsistentUri);
+            DeletePOptions deleteOptions = DeletePOptions.newBuilder().setAlluxioOnly(true).build();
+            mFileSystem.delete(inconsistentUri, deleteOptions);
+            mFileSystem.exists(inconsistentUri);
+            System.out.println(inconsistentUri + " repaired");
+            System.out.println();
+          } catch (AlluxioException | IOException e) {
+            exceptions.add(e);
+          }
+        }, true);
       }
+
+      waitForTasks(completionService, totalUris, exceptions);
```

Comment:
should we print out that we are waiting for everything to finish?

Reply:
I don't think there's a need to. This is more or less just the code making sure we do files before directories. While this is "waiting" there will be plenty of tasks running on background threads which are printing to the CLI. To them they won't be able to tell that anything is "waiting". They'll probably end up seeing some out-of-order log messages